From 1a31c35732db748af7af8d7d123d148582376251 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Wed, 9 Nov 2022 15:28:06 +0000
Subject: [PATCH] trivial: Fix the tests on s390x

---
 plugins/synaptics-mst/fu-self-test.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/plugins/synaptics-mst/fu-self-test.c b/plugins/synaptics-mst/fu-self-test.c
index ffdaa7a5b..29a05e9c5 100644
--- a/plugins/synaptics-mst/fu-self-test.c
+++ b/plugins/synaptics-mst/fu-self-test.c
@@ -81,6 +81,11 @@ fu_plugin_synaptics_mst_none_func(void)
 	    g_ptr_array_new_with_free_func((GDestroyNotify)g_object_unref);
 	g_autofree gchar *filename = NULL;
 
+#if defined(__s390x__)
+	g_test_skip("Skipping HWID test on s390x due missing HwIDs");
+	return;
+#endif
+
 	ret = fu_context_load_quirks(ctx, FU_QUIRKS_LOAD_FLAG_NO_CACHE, &error);
 	g_assert_no_error(error);
 	g_assert_true(ret);
@@ -124,6 +129,11 @@ fu_plugin_synaptics_mst_tb16_func(void)
 	    g_ptr_array_new_with_free_func((GDestroyNotify)g_object_unref);
 	g_autofree gchar *filename = NULL;
 
+#if defined(__s390x__)
+	g_test_skip("Skipping HWID test on s390x due missing HwIDs");
+	return;
+#endif
+
 	ret = fu_context_load_quirks(ctx, FU_QUIRKS_LOAD_FLAG_NO_CACHE, &error);
 	g_assert_no_error(error);
 	g_assert_true(ret);
-- 
2.37.3

