diff --git a/meson.build b/meson.build
index da72f33f..73420d18 100644
--- a/meson.build
+++ b/meson.build
@@ -77,7 +77,7 @@ warning_flags = [
   '-Wmissing-prototypes',
   '-Wnested-externs',
   '-Wno-cast-function-type',
-  '-Wno-error=cpp',
+  '-Wno-address-of-packed-member',  # incompatable with g_autoptr()
   '-Wno-unknown-pragmas',
   '-Wno-discarded-qualifiers',
   '-Wno-missing-field-initializers',
diff --git a/plugins/uefi/efi/meson.build b/plugins/uefi/efi/meson.build
index dbccda03..5be2ffea 100644
--- a/plugins/uefi/efi/meson.build
+++ b/plugins/uefi/efi/meson.build
@@ -53,7 +53,6 @@ compile_args = ['-Og',
                 '--param=ssp-buffer-size=4',
                 '-fexceptions',
                 '-Wall',
-                '-Werror',
                 '-Wextra',
                 '-Wvla',
                 '-std=gnu11',
@@ -66,10 +65,14 @@ compile_args = ['-Og',
                 '-fno-merge-constants',
                 '-Wsign-compare',
                 '-Wno-missing-field-initializers',
+                '-Wno-address-of-packed-member',
                 '-grecord-gcc-switches',
                 '-DDEBUGDIR="@0@"'.format(debugdir),
                 '-isystem', efi_incdir,
                 '-isystem', join_paths(efi_incdir, gnu_efi_path_arch)]
+if get_option('werror')
+    compile_args += '-Werror'
+endif
 if efi_arch == 'x86_64'
   compile_args += ['-mno-red-zone',
                    '-mno-sse',
commit 96e67839822d87be94372e204f24d9bf8672f4fb
Author: Richard Hughes <richard@hughsie.com>
Date:   Fri Feb 1 16:47:03 2019 +0000

    trivial: Fix potential compile failures for high -j values

diff --git a/plugins/ata/meson.build b/plugins/ata/meson.build
index cd26e395..7171e6b4 100644
--- a/plugins/ata/meson.build
+++ b/plugins/ata/meson.build
@@ -36,6 +36,7 @@ if get_option('tests')
   cargs += '-DTESTDATADIR="' + testdatadir + '"'
   e = executable(
     'ata-self-test',
+    fu_hash,
     sources : [
       'fu-self-test.c',
       'fu-ata-device.c',
diff --git a/plugins/dell/meson.build b/plugins/dell/meson.build
index df96b979..2c0532f1 100644
--- a/plugins/dell/meson.build
+++ b/plugins/dell/meson.build
@@ -37,6 +37,7 @@ if get_option('tests')
   cargs += '-DPLUGINBUILDDIR="' + meson.current_build_dir() + '"'
   e = executable(
     'dell-self-test',
+    fu_hash,
     sources : [
       'fu-self-test.c',
       'fu-dell-smi.c',
diff --git a/plugins/dfu/meson.build b/plugins/dfu/meson.build
index f9d9632d..fc517fcc 100644
--- a/plugins/dfu/meson.build
+++ b/plugins/dfu/meson.build
@@ -6,6 +6,7 @@ install_data(['dfu.quirk'],
 
 dfu = static_library(
   'dfu',
+  fu_hash,
   sources : [
     'dfu-cipher-xtea.c',
     'dfu-common.c',
@@ -117,6 +118,7 @@ if get_option('tests')
   cargs += '-DTESTDATADIR="' + testdatadir + '"'
   e = executable(
     'dfu-self-test',
+    fu_hash,
     sources : [
       'dfu-self-test.c'
     ],
diff --git a/plugins/nitrokey/meson.build b/plugins/nitrokey/meson.build
index 46911498..225b49b7 100644
--- a/plugins/nitrokey/meson.build
+++ b/plugins/nitrokey/meson.build
@@ -30,6 +30,7 @@ shared_module('fu_plugin_nitrokey',
 if get_option('tests')
   e = executable(
     'nitrokey-self-test',
+    fu_hash,
     sources : [
       'fu-nitrokey-common.c',
       'fu-self-test.c',
diff --git a/plugins/nvme/meson.build b/plugins/nvme/meson.build
index 614ccc60..5ec0a862 100644
--- a/plugins/nvme/meson.build
+++ b/plugins/nvme/meson.build
@@ -38,6 +38,7 @@ if get_option('tests')
   cargs += '-DTESTDATADIR="' + testdatadir + '"'
   e = executable(
     'nvme-self-test',
+    fu_hash,
     sources : [
       'fu-self-test.c',
       'fu-nvme-common.c',
diff --git a/plugins/redfish/meson.build b/plugins/redfish/meson.build
index a4cdba55..5c88504e 100644
--- a/plugins/redfish/meson.build
+++ b/plugins/redfish/meson.build
@@ -32,6 +32,7 @@ install_data(['redfish.conf'],
 if get_option('tests')
   e = executable(
     'redfish-self-test',
+    fu_hash,
     sources : [
       'fu-self-test.c',
       'fu-redfish-client.c',
diff --git a/plugins/synapticsmst/meson.build b/plugins/synapticsmst/meson.build
index 0f1abdbf..046ac6e4 100644
--- a/plugins/synapticsmst/meson.build
+++ b/plugins/synapticsmst/meson.build
@@ -35,6 +35,7 @@ if get_option('tests')
   cargs += '-DSOURCEDIR="' + meson.current_source_dir() + '"'
   e = executable(
     'synapticsmst-self-test',
+    fu_hash,
     sources : [
       'fu-self-test.c',
       'synapticsmst-common.c',
diff --git a/plugins/thunderbolt/meson.build b/plugins/thunderbolt/meson.build
index 10991e5b..c15e4d87 100644
--- a/plugins/thunderbolt/meson.build
+++ b/plugins/thunderbolt/meson.build
@@ -26,6 +26,7 @@ testdatadir_src = join_paths(meson.source_root(), 'data', 'tests')
 testdatadir_dst = join_paths(meson.build_root(), 'data', 'tests')
 cargs += '-DTESTDATADIR="' + testdatadir_src + ':' + testdatadir_dst + '"'
 executable('tbtfwucli',
+  fu_hash,
   sources : [
     'fu-thunderbolt-tool.c',
   ],
@@ -50,6 +51,7 @@ if get_option('tests') and umockdev.found() and gio.version().version_compare('>
   cargs += '-DPLUGINBUILDDIR="' + meson.current_build_dir() + '"'
   e = executable(
     'thunderbolt-self-test',
+    fu_hash,
     sources : [
       'fu-self-test.c',
       'fu-plugin-thunderbolt.c',
diff --git a/plugins/udev/meson.build b/plugins/udev/meson.build
index 334b14a8..dfc05def 100644
--- a/plugins/udev/meson.build
+++ b/plugins/udev/meson.build
@@ -50,6 +50,7 @@ if get_option('tests')
   cargs += '-DTESTDATADIR="' + testdatadir + '"'
   e = executable(
     'udev-self-test',
+    fu_hash,
     sources : [
       'fu-self-test.c',
       'fu-rom.c',
diff --git a/plugins/uefi/meson.build b/plugins/uefi/meson.build
index 2492e4ec..fd1b3976 100644
--- a/plugins/uefi/meson.build
+++ b/plugins/uefi/meson.build
@@ -83,6 +83,7 @@ if get_option('tests')
   cargs += '-DTESTDATADIR="' + testdatadir + '"'
   e = executable(
     'uefi-self-test',
+    fu_hash,
     sources : [
       'fu-self-test.c',
       'fu-uefi-bgrt.c',
diff --git a/plugins/unifying/meson.build b/plugins/unifying/meson.build
index 7e04912d..f90d1fbf 100644
--- a/plugins/unifying/meson.build
+++ b/plugins/unifying/meson.build
@@ -39,6 +39,7 @@ shared_module('fu_plugin_unifying',
 if get_option('tests')
   e = executable(
     'unifying-self-test',
+    fu_hash,
     sources : [
       'fu-unifying-self-test.c',
       'fu-unifying-common.c',
diff --git a/plugins/wacom-usb/meson.build b/plugins/wacom-usb/meson.build
index e2a3f37c..fd92ed4b 100644
--- a/plugins/wacom-usb/meson.build
+++ b/plugins/wacom-usb/meson.build
@@ -38,6 +38,7 @@ if get_option('tests')
   cargs += '-DTESTDATADIR="' + testdatadir + '"'
   e = executable(
     'wacom-usb-self-test',
+    fu_hash,
     sources : [
       'fu-self-test.c',
       'fu-wac-common.c',
